WebKit/WebKit
# Build fix: add gtest dependency to gmock project

#### 14adb9294c87dee26eddf41b4dcdc07c50389fd1
<pre>
Build fix: add gtest dependency to gmock project
<a href="https://bugs.webkit.org/show_bug.cgi?id=263448">https://bugs.webkit.org/show_bug.cgi?id=263448</a>
&lt;rdar://117266314&gt;

Reviewed by Elliott Williams.

Summary of changes:
- To fix the gmock build, add a &quot;fake&quot; dependency to the gmock target to
  libgtest.a by creating a &quot;Copy Files&quot; build phase script named
  &quot;Product Dependencies&quot;.  This mimics what the WTF project does for
  libbmalloc.a and libWebKitAdditions.a.
- Add two &quot;Copy Files&quot; build phase scripts to the gtest-static target so
  that internal and internal/custom headers are copied into
  $(BUILT_PRODUCTS_DIR)/usr/local/include/gtest.  We can eventually
  remove gtest-framework since nothing links to it--it is only used for
  including headers.
- Remove the gmock_main target added in 269490@main since it is not
  currently used.

* Source/ThirdParty/gmock/Configurations/gmock_main.xcconfig: Remove.
* Source/ThirdParty/gmock/gmock.xcodeproj/project.pbxproj:
- Remove gtest_main target and xcconfig file.
- Add &quot;Product Dependencies&quot; build phase script to create a dependency
  on gtest-static when building from a workspace.  This fixes the build.
* Source/ThirdParty/gtest/xcode/gtest.xcodeproj/project.pbxproj:
- Install internal and internal/custom headers with gtest-static target
  into $(BUILT_PRODUCTS_DIR)/usr/local/include/gtest.
- Remove project headers for gtest-framework target since internal and
  internal/custom paths are not preserved, and gtest.h is a duplicate
  name.

Canonical link: <a href="https://commits.webkit.org/269581@main">https://commits.webkit.org/269581@main</a>
</pre>

https://github.com/WebKit/WebKit/commit/16105069a7999f75acfac9680e6b47ec93e5fff3

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/22976 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/1054 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/24091 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/24890 "Built successfully") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/32/builds/21265 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/2280 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/23517 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/22138 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/23218 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/47/builds/517 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/19935 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/25747 "Built successfully") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/42/builds/405 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/20817 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/27006 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/20779 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/21078 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/24865 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/44/builds/502 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/17/builds/18302 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/420 "Built successfully") | | | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/43/builds/910 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/2910 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/680 "Built successfully") | | |
